Dockerfile

```diff
@@ -17,7 +17,11 @@ RUN make juno
 RUN upx /app/build/juno
 
 # Stage 2: Build Docker image
-FROM alpine:3.14 AS runtime
+FROM alpine:3.18 AS runtime
+
+# Install runtime dependencies
+RUN apk update --no-cache && \
+    apk add --no-cache libgcc
```

Comment:
I just noticed that we install `libgcc` in runtime. Why do we need it here?

Reply:
Not needed, removed

Reply:
My bad, we do need it.

https://github.com/NethermindEth/juno/actions/runs/5405966720/jobs/9822237984?pr=850

```
2023/06/28 21:38:06 Node log: Error loading shared library libgcc_s.so.1: No such file or directory (needed by /usr/local/bin/juno)
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_Resume: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_GetIPInfo: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_GetDataRelBase: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_GetRegionStart: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_SetGR: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_FindEnclosingFunction: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_GetTextRelBase: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_DeleteException: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_RaiseException: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_GetIP: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_Backtrace: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_GetCFA: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_GetLanguageSpecificData: symbol not found
2023/06/28 21:38:06 Node log: Error relocating /usr/local/bin/juno: _Unwind_SetIP: symbol not found
```

Reply:
so libgcc linked as shared library?

Reply:
Yeap, unless you tell the linker to link a library statically it will use dynamic linkage by default.